internal/tfdiags/hcl.go

```diff
@@ -57,6 +57,24 @@ func (d hclDiagnostic) ExtraInfo() interface{} {
 	return d.diag.Extra
 }
 
+func (d hclDiagnostic) Equals(otherDiag ComparableDiagnostic) bool {
```

Comment:
This can compare diagnostics with the same text from different locations as equal, when they should be all enumerated separately.